### PR TITLE
Add Drone CI and dapperize

### PR DESCRIPTION
**Pull request by superseb**

### CODE REVIEW

File: scripts/test [hunk 1]
```diff
@@ -1,20 +1,29 @@
...
+
+while true; do
+  docker version && break
+  sleep 5
+done
 
 echo "==> Testing install-docker scripts..."
 
-# https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string
-SEMVER_REGEX='^(\d*)\.(\d*)\.(\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$'
```
Comment (rawmind0) [inline scripts/test:8]: Why `SEMVER_REGEX` is removed?? It was here to assure that just `pkg/<SEMVER>` are processed.

File: scripts/test [hunk 2]
```diff
@@ -1,20 +1,29 @@
...
+cd pkg
 
-cd $(dirname $0)/../pkg
```
Comment (rawmind0) [inline scripts/test:10]: Is not better to use relative path within repo?? It should work in a dapper or not dapper env

File: scripts/test [hunk 3]
```diff
@@ -26,19 +35,31 @@ for DOCKER_VERSION in $(ls -1d */ | cut -d/ -f1 | grep -E $SEMVER_REGEX ); do
...
-				COMMAND="sh /install.sh"
+				COMMAND="bash /install.sh"
 				# Patch for oracle linux 8 docker image as yum is not installed
 				if [ "$IMAGE" == "oraclelinux" ] && [ "$TAG" == "8" ]; then
-					COMMAND="dnf install -y yum && sh /install.sh"
+					COMMAND="dnf install -y yum && bash /install.sh"
 				fi
 				docker run --rm -it --privileged \
-					-v $PWD/../dist/$DOCKER_VERSION.sh:/install.sh \
+					-v /install-docker/dist/$DOCKER_VERSION.sh:/install.sh \
```
Comment (rawmind0) [inline scripts/test:44]: Is not better to use relative path within repo?? It should work in a dapper or not dapper env

File: scripts/test [hunk 4]
```diff
@@ -1,20 +1,29 @@
...
 
 echo "==> Testing install-docker scripts..."
 
-# https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string
-SEMVER_REGEX='^(\d*)\.(\d*)\.(\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$'
+cd pkg
 
-cd $(dirname $0)/../pkg
-for DOCKER_VERSION in $(ls -1d */ | cut -d/ -f1 | grep -E $SEMVER_REGEX ); do
+for DOCKER_VERSION in $(ls -1 .); do
```
Comment (rawmind0) [inline scripts/test:19]: Why SEMVER_REGEX is removed?? It was here to assure that just `pkg/<SEMVER>` are processed.

File: scripts/test (hunk 1 above)
Comment (superseb) [reply scripts/test:8]: What is the situation where a non semver ends up in `pkg/` and this check is needed?

Comment (rawmind0) [reply scripts/test:8]: I think this check is needed as we are releasing SEMVER notation named scripts, `pkg` are building `dist` files. I don't think is a bad practice to check that naming is using SEMVER notation. What's the reason to remove it?? 

Anyway, if `SEMVER_REGEX ` filter is removed, the `README.md` should also be updated as https://github.com/rancher/install-docker#add-new-docker-release is saying `...version should follow semver format...`

File: scripts/validate [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+for DOCKER_VERSION in $(ls -1 ../pkg/); do
+    SEMVER=$(chsv_check_version $DOCKER_VERSION)
+    if [ -z $SEMVER ]; then
+        echo "[ERR] Version found in pkg directory is not semver (${DOCKER_VERSION})"
+        exit 1
+    fi
+done
+
+echo "Checking basic shell validation"
+find ../dist -name \*.sh -exec sh -n {} \;
```
Comment (rawmind0) [inline scripts/validate:26]: May be would be better to add `check_semversion`  as bash function and could be called for both paths, `check_semversion pkg/` and `check_semversion dist/`

```
function check_semversion() {
    if [ -z $1 ]; then
        echo "[ERR] Checking sameversion: directory to check is empty"
        exit 1
    fi
    if [ ! -d $1 ]; then
        echo "[ERR] Checking sameversion in $1 directory: directory doesn't exist"
        exit 1
    fi
    for DOCKER_VERSION in $(ls -1 $1); do
        SEMVER=$(chsv_check_version $DOCKER_VERSION)
        if [ -z $SEMVER ]; then
            echo "[ERR] Version found in $1 directory is not semver (${DOCKER_VERSION})"
            exit 1
        fi
    done
}
```

File: .drone.yml [hunk 6]
```diff
@@ -0,0 +1,43 @@
+---
+kind: pipeline
+name: amd64
+
+platform:
+  os: linux
+  arch: amd64
+
+steps:
+  - name: validate
```
Comment (rawmind0) [inline .drone.yml:10]: I think this validate should check if `dist/` directory contains only semver compatible version, but the `scripts/validate` is just checking for `pkg/` directory.

Comment (rawmind0) [inline .drone.yml:10]: Once `validate` is already included at `test` script the `validate` step wouldn't be required any more, is it??

**Review by rawmind0 — APPROVED** [review]
LGTM

**Pull request merged**